packages/client/hmi-client/src/components/models/tera-stratify-output-model-diagram.vue

```diff
@@ -0,0 +1,205 @@
+					<div v-if="model" ref="graphElement" class="graph-element" />
+				</section>
+			</div>
+		</TeraResizablePanel>
+	</main>
+</template>
+
+<script setup lang="ts">
+import { IGraph } from '@graph-scaffolder/index';
+import { watch, ref, computed } from 'vue';
+import { runDagreLayout } from '@/services/graph';
+import {
+	NodeData,
+	EdgeData,
+	PetrinetRenderer
+} from '@/model-representation/petrinet/petrinet-renderer';
+import { NestedPetrinetRenderer } from '@/model-representation/petrinet/nested-petrinet-renderer';
+import { convertToIGraph } from '@/model-representation/petrinet/petrinet-service';
+import { Model } from '@/types/Types';
+import { useNodeTypeColorPalette } from '@/utils/petrinet-color-palette';
+import Button from 'primevue/button';
+import Toolbar from 'primevue/toolbar';
+import TeraResizablePanel from '../widgets/tera-resizable-panel.vue';
+
+const props = defineProps<{
+	model: Model;
+}>();
+
+const graphElement = ref<HTMLDivElement | null>(null);
+let renderer: PetrinetRenderer | null = null;
+
+const stateTypes = computed(() => {
+	if (props.model.semantics?.typing?.system.states) {
+		return props.model.semantics.typing.system.states.map((s) => s.name);
+	}
+	return props.model.semantics?.typing?.system?.model.states.map((s) => s.name);
+});
+const transitionTypes = computed(() => {
+	if (props.model.semantics?.typing?.system.transitions) {
+		return props.model.semantics.typing.system.transitions.map((t) => t.properties?.name);
+	}
+	return props.model.semantics?.typing?.system?.model.transitions.map((t) => t.properties?.name);
+});
+
+const isCollapsed = ref(true);
+
+const { getNodeTypeColor } = useNodeTypeColorPalette();
+
+function getLegendKeyStyle(id: string) {
+	return {
+		backgroundColor: getNodeTypeColor(id)
+	};
+}
+function toggleCollapsedView() {
+	isCollapsed.value = !isCollapsed.value;
+}
+
+// Render graph whenever a new model is fetched or whenever the HTML element
+//	that we render the graph to changes.
+watch(
+	[() => props.model, graphElement, isCollapsed],
+	async () => {
+		if (props.model === null || graphElement.value === null) return;
+		const graphData: IGraph<NodeData, EdgeData> = convertToIGraph(
+			isCollapsed.value ? props.model.semantics?.span?.[0].system : props.model
+		);
+
+		const nestedMap = props.model.semantics?.span?.[0].map.reduce(
+			(childMap, [stratNode, baseNode]) => {
+				if (!childMap[baseNode]) {
+					childMap[baseNode] = [];
+				}
+				childMap[baseNode].push(stratNode);
+				return childMap;
+			},
+			{}
+		);
+		// Create renderer
+		if (!renderer) {
+			renderer = new NestedPetrinetRenderer({
+				el: graphElement.value as HTMLDivElement,
+				useAStarRouting: false,
+				useStableZoomPan: true,
+				runLayout: runDagreLayout,
+				dragSelector: 'no-drag',
+				nestedMap
+			});
+		} else {
+			renderer.isGraphDirty = true;
+		}
+
+		// Render graph
+		await renderer?.setData(graphData);
+		await renderer?.render();
+	},
+	{ deep: true }
```

Comment:
Do we really need deep watchers here?  
Slightly concerned because `props.model` is not something you want to run recursive watchers on.

Reply:
This is the same for `tera-model-diagram` and all its derivatives. However, for this component we don't need to watch `props.model` at all as it shouldn't change.

Reply:
`tera-model-diagram` is not a good component to model after, it has a lot unnecessary baggages from a incomplete refactor so nearly all it's states, reactive elements, watchers and computed values are questionable.